### ISSUE TITLE
Naming/UncommunicativeMethodParamName doesn't take into account underscore prefix

### ISSUE THREAD

**Issue by thomthom** [rubocop-hq/rubocop#6761]
I use Ruby in the domain of 3D modelling. So common short variables include `x`, `y`, `z` and `u`, `v`. These are flagged by the default rules of `Naming/UncommunicativeMethodParamName` so I add them to the list of allowed names:

```yml
Naming/UncommunicativeMethodParamName:
  AllowedNames:
    - x
    - y
    - z
    - u
    - v
    - uv
```

However, some times these names are also used as argument names in callbacks of the API. And you don't always use all the arguments being passed in the callback. For instance:

```ruby
 def onLButtonDoubleClick(_flags, _x, _y, _view)
  # ...
end
```

However this then will again flag the `x` and `y` arguments:

```
src/tt_truebend/tools/truebend.rb:195:42: C: Naming/UncommunicativeMethodParamName: Method parameter must be at least 3 characters long.
    def onLButtonDoubleClick(_flags, _x, _y, _view)
```

I could add `_x` and `_y` to the list of allowed names. But this still feels like a bug - where the cop should take into account the convention of using underscore prefix to denote unused arguments.

--------

## Expected behavior

I would expect `Naming/UncommunicativeMethodParamName` to take into account the `_` prefix convention and strip it away when it checks against the list of allowed short names.

## Actual behavior

Currently `_x` is flagged even if `x` is an allowed variable name.

## Steps to reproduce the problem

```yml
Naming/UncommunicativeMethodParamName:
  AllowedNames:
    - x
    - y
    - z
```

```ruby
 def onLButtonDoubleClick(_flags, _x, _y, _view)
  # ...
end
```

## RuboCop version

```
$ [bundle exec] rubocop -V
0.63.1 (using Parser 2.6.0.0, running on ruby 2.2.6 x64-mingw32)
```

**Comment by Drenmi** [comment]
I think this sounds like a perfectly reasonable enhancement to the cop. 🙂 Interested in making a PR @thomthom?

**Comment by thomthom** [comment]
Yea, I can make a PR when I have a spare moment.

**Comment by thomthom** [comment]
Starting work on this now.
